test/consistent-destructuring.js

```diff
@@ -344,6 +344,26 @@ ruleTester.run('consistent-destructuring', rule, {
 				const {a} = foo;
 				console.log(!a);
 			`
+		}),
+		invalidTestCase({
+			code: outdent`
+				const {a, ...b} = foo;
+				console.log(foo.c);
+			`,
+			correctCode: outdent`
+				const {a, c, ...b} = foo;
+				console.log(c);
+			`
+		}),
+		invalidTestCase({
+			code: outdent`
+				const {a, ...b} = foo;
+				console.log(foo.b);
+			`,
+			correctCode: outdent`
+				const {a, b: b_, ...b} = foo;
```

Comment:
You can't do this,  value of `b` changed.

If contains `RestElement`, it should not fixable, and should be correct code.

Reply:
Except for cases where the member is already destructured, right?

Reply:
I don't think we need check destructing with RestElement at all, it' not changable. It's correct code.

`const {a, ...b} = foo;` no better/equal way.

We need ignore it unless `b` is destructing again.


# Valid no error

```js
const {a, ...b} = foo;
alert(foo.c)
```
# Invalid

```js
const {a, ...b} = foo;
const {d}=b;
alert(b.c)
```

Reply:
Sorry, I fogot we need check `foo.a` , I'll review tomorrow, on cellphone.

Reply:
> const {a, ...b} = foo; no better/equal way.

Yeah, except for the case where somebody uses `foo.a`. We can fix that to `a` without changing the object expression at all.